core/src/main/scala/org/apache/spark/sql/delta/sources/DeltaSource.scala

```diff
@@ -107,6 +108,15 @@ trait DeltaSourceBase extends Source
 
   protected var lastOffsetForTriggerAvailableNow: DeltaSourceOffset = _
 
+  override def prepareForTriggerAvailableNow(): Unit = {
```

Comment:
I think it will be good reuse all the existing code paths for finding the next offset. 
I see that we already make sure in `AdmissionLimits.apply` returns None when ReadLimits.availableNow() is used. And when the limit is None, then `getFileChangesWithRateLimit` already skipping filtering the AddFile actions.

I believe the code will become something like `latestOffset(null, ReadLimit.allAvailable())`

Reply:
Hmm there's some weird stuff happening when I try to do that, it looks like weird conflicts between the legacy `org.apache.spark.sql.execution.streaming.Offset` and the new `org.apache.spark.sql.connector.read.streaming.Offset`. `latestOffset` returns the latter, but everything else in the DeltaSource uses the former. When I try to update everything to use the newer version, the tests just fail with no data being read. Still trying to figure it out

Reply:
Oh.. then we should just try to convert it. something like `Option(latestOffset(null, ReadLimit.allAvailable())).map(o => DeltaSourceOffset(tableId, o.asInstanceOf[Offset])).orNull` ?

Reply:
Figured out I just need to tweak the `DeltaSourceOffset.apply` to accept the connector Offset

Reply:
Great!